### ISSUE TITLE
'for' limit must be a number 

### ISSUE THREAD

**Issue by gorillatarzan** [RevTheDev/rewards#6]
I am facing a problem on random rewards.
only one player can take the reward... and then an error appears ...
![image](https://user-images.githubusercontent.com/59956801/121817192-8beb3c00-cc88-11eb-98a3-0753ddb10616.png)
any help on this ? 
Thank you very much!

**Comment by RevTheDev** [comment]
Please double check the config file in the rewards folder and check the limit is a number also please make sure you haven't renamed the folder

**Comment by RevTheDev** [comment]
Actually scrap that iv just noticed the unknown error for your MySQL it is a database problem one or more items may be missing from your database or a table is missing double check with you database and the rewards config that the random items it gives are indeed in your items area in your database

**Comment by gorillatarzan** [comment]
I check all the items and everything is added to sql... I even remove everyhting and i put only water and bread... Only one player can take the reward... after first player get the reward then if another player go , i get this message .
please see screenshots.
![image](https://user-images.githubusercontent.com/59956801/122054258-2ca83b80-cdf0-11eb-8f4b-55211ee987ff.png)
![image](https://user-images.githubusercontent.com/59956801/122054389-4cd7fa80-cdf0-11eb-89eb-27e31f0bc38b.png)

**Comment by gorillatarzan** [comment]
Screen shot of the line 63
![image](https://user-images.githubusercontent.com/59956801/122054945-cff95080-cdf0-11eb-92db-b6aaec370645.png)

**Comment by RevTheDev** [comment]
If you can try enabling random rewards and see if that works then if it does it might be something to do with the way items are sort but like I say I can't really identify stuff atm as I'm rebuilding my pc

**Comment by RevTheDev** [comment]
Also the best bet is to probably remove rewards from your server and delete the section for rewards in your server database and redownload the master file of GitHub and run it as is without changing anything add the sql to the database database test it if it still does it then you'll know something is wrong on your server or database

**Comment by gorillatarzan** [comment]
I put the script as it is ... and i have the same issue

**Comment by gorillatarzan** [comment]
if i remove the line that is show on the screen shot random is working fine .. but is not getting the chances ... I mean is giving every item with the same percentage random ....
So 20 or 50 % is the same for the code... is just giving random 
![image](https://user-images.githubusercontent.com/59956801/122100349-bf120480-ce1b-11eb-8fbb-3628edbb982b.png)

**Comment by RevTheDev** [comment]
Have you tried disabling random rewards coz that bit of code is only for the random rewards

**Comment by gorillatarzan** [comment]
Yes when i remove the random rewards all works perfect …  So something is going wrong with the specific line

**Comment by RevTheDev** [comment]
Okay I'll have a look through

**Comment by gorillatarzan** [comment]
Ok thank you very much and please inform me for any news :D

**Comment by RevTheDev** [comment]
just tested it on my test server and its working perfectly i cant get it to duplicate your error

**Comment by gorillatarzan** [comment]
i dont know what is going wrong... something  is going wrong with the table.... 
i google it a little bit and i can see that the original version have the same issue 
https://github.com/Elipse458/el_daily_rewards/commit/3e531f96807acff5dd3af2dc9168a0084fc4e726

**Comment by RevTheDev** [comment]
The only thing I can think of is go into your server and try /restart rewards or try a full server restart and clear your server cache

**Comment by gorillatarzan** [comment]
I already try it ... nothing... I don't know what is going wrong... Anyway thank you for your time

**Comment by Bluxrs** [comment]
I think its because the two config items you have, have the same chance number (80). If you change one of them to 81 you will not get any errors anymore but it will most likely not be doing what you want it to do so you will need to do the below.

I wanted to have it be the same number so it would give them either items randomly but you will need to rewrite some of the code to do this with a random chance of 80, then pull from array (list of items).
